course_catalog/views.py

```diff
@@ -466,3 +466,22 @@ class PodcastEpisodesViewSet(viewsets.ReadOnlyModelViewSet):
             Prefetch("topics", queryset=CourseTopic.objects.all()),
         )
     )
+
+
+class EpisodesInPodcast(viewsets.ReadOnlyModelViewSet):
+    """
+    Viewset for listing PodcastEpisodes for a given Podcast primary key
+    """
+
+    serializer_class = PodcastEpisodeSerializer
+    permission_classes = (ReadOnly & PodcastFeatureFlag,)
+
+    def get_queryset(self):
+        return (
+            PodcastEpisode.objects.filter(published=True, podcast=self.kwargs["pk"])
```

Comment:
This may be an unrealistic edge case, but if I have published episodes in an unpublished podcast, I get the list of the published episodes returned, whereas the API URL for the podcast itself returns a 404.   Maybe that's more of a bad data situation?

Reply:
I added a filter so that an unpublished podcast will return an empty list